components/dashboard/src/workspaces/ConnectToSSHModal.tsx

```diff
@@ -0,0 +1,147 @@
+    };
+    const tip = props.tip ?? "Click to copy";
+    return (
+        <div className={`w-full relative ${props.className ?? ""}`}>
+            <input
+                disabled={true}
+                readOnly={true}
+                autoFocus
+                className="w-full pr-8 overscroll-none"
+                type="text"
+                defaultValue={props.value}
+            />
+            <div className="cursor-pointer" onClick={() => copyToClipboard(props.value)}>
+                <div className="absolute top-1/3 right-3">
+                    <Tooltip content={copied ? "Copied!" : tip}>
+                        <img src={copy} alt="copy icon" title={tip} />
+                    </Tooltip>
+                </div>
+            </div>
+        </div>
+    );
+}
+
+interface SSHProps {
+    workspaceId: string;
+    ownerToken: string;
+    ideUrl: string;
+}
+
+function SSHManualView(props: SSHProps) {
+    return (
+        <>
+            <div className="mt-1 mb-2">
+                <p className="text-gray-500 whitespace-normal text-base">
+                    The following args can be used with SSH to connect to this workspace.
+                </p>
+            </div>
+            <div className="flex flex-col">
+                <div className="mt-2">
+                    <h4>Host</h4>
+                    <InputWithCopy value={props.ideUrl} />
+                </div>
+                <div className="mt-2">
+                    <h4>Username</h4>
+                    <InputWithCopy value={props.workspaceId} />
+                </div>
+                <div className="mt-2">
+                    <h4>Password</h4>
+                    <InputWithCopy value={props.ownerToken} />
+                </div>
+            </div>
+        </>
+    );
+}
+
+function SSHPassView(props: SSHProps) {
+    const sshCommand = `sshpass -p ${props.ownerToken} ssh -o StrictHostKeyChecking=no ${props.workspaceId}@${props.ideUrl}`;
+    return (
+        <>
+            <div className="mt-1 mb-2">
+                <p className="text-gray-500 whitespace-normal text-base">
+                    The following shell command can be used to SSH into this workspace.
+                </p>
+            </div>
+            <InputWithCopy value={sshCommand} tip="Copy SSH Command" />
+            <div className="mt-2">
+                DOCUMENT ABOUT <strong>sshpass</strong> NEEDED
+            </div>
+        </>
+    );
+}
+
+export default function ConnectToSSHModal(props: {
+    workspaceId: string;
+    ownerToken: string;
+    ideUrl: string;
+    onClose: () => void;
+}) {
+    interface MenuItem {
+        key: string;
+        component: React.ReactChild;
+    }
+    const menuList: MenuItem[] = [
+        {
+            key: "manual",
+            component: (
+                <SSHManualView workspaceId={props.workspaceId} ownerToken={props.ownerToken} ideUrl={props.ideUrl} />
+            ),
+        },
+        {
+            key: "sshpass",
```

Comment:
**issue**: While this seems efficient do we really want to promote tools like this? Are you regularly using this, @iQQBot? I could be wrong but I assume many users will be against installing tools like this simply to use SSH access.

Reply:
Wow, that's a neat hack :)

It is also a bit indirect and harder to understand.
Frankly I'd vote for going for the manual/sshpass options as they stand and make sure we have really good documentation for both cases in place.